selendroid-standalone/src/main/java/io/selendroid/standalone/android/AndroidSdk.java

```diff
@@ -124,25 +108,91 @@ public static String androidJar() {
   }
 
   public static File androidSdkFolder() {
-    String platformsRootFolder = androidHome() + File.separator + "platforms";
-    File platformsFolder = new File(platformsRootFolder);
-
     if (sAndroidSdkVersion != null) {
-      return new File(platformsFolder, sAndroidSdkVersion);
-    } else {
-      return findLatestAndroidPlatformFolder(
-        platformsFolder,
-        "No installed Android APIs have been found.");
+      return new File(platformsFolder(), sAndroidSdkVersion);
+    }
+    return findLatestAndroidPlatformFolder();
+  }
+
+  private static File platformsFolder() {
+    return new File(androidHome(), "platforms");
+  }
+
+  public static File findLatestAndroidPlatformFolder() {
+    return findLatestAndroidPlatformFolder(platformsFolder());
+  }
+
+  public static File findLatestAndroidPlatformFolder(File platformsFolder) {
+    File[] platformVersions = platformsFolder.listFiles(new FileFilter() {
+      @Override
+      public boolean accept(File file) {
+        return file.getName().matches("android-\\d+$");
+      }
+    });
+
+    if (platformVersions == null || platformVersions.length == 0) {
+      throw new SelendroidException("No valid Android platform folder found in " + platformsFolder.getName());
+    }
+
+    Arrays.sort(
+      platformVersions,
+      new Comparator<File>() {
+        @Override
+        public int compare(File f1, File f2) {
+          int v1 = Integer.parseInt(f1.getName().split("-")[1]);
+          int v2 = Integer.parseInt(f2.getName().split("-")[1]);
+          return Integer.compare(v1, v2);
+        }
+      });
+
+    return platformVersions[platformVersions.length - 1].getAbsoluteFile();
+  }
+
+  public static File buildToolsFolder() {
+    if (sBuildToolsVersion != null) {
+      return new File(buildToolsHome(), sBuildToolsVersion);
     }
+    return findLatestBuildToolsFolder();
+  }
+
+  public static File findLatestBuildToolsFolder() {
+    return findLatestBuildToolsFolder(buildToolsHome());
   }
 
-  protected static File findLatestAndroidPlatformFolder(File rootFolder, String errorMessage) {
-    File[] androidApis = rootFolder.listFiles(new AndroidFileFilter());
-    if (androidApis == null || androidApis.length == 0) {
-      throw new SelendroidException(errorMessage);
+  public static File findLatestBuildToolsFolder(File buildToolsHome) {
+    File[] buildToolsVersions = buildToolsHome.listFiles(new FileFilter() {
+      @Override
+      public boolean accept(File file) {
+        return file.getName().matches("\\d+\\.\\d+\\.\\d+$");
```

Comment:
Similar comment to above re: it would be cleaner if this regular expression had capture groups and was reused on lines 179-180 instead of splitting the string.